### DISCUSSION TITLE
Add possibility to build multiple Directus extensions from one package

### DISCUSSION THREAD

**Discussion by d1rOn** [directus/directus#12163] [Feature Requests]
As for now we should to create different packages for different Directus extensions.
But sometimes extensions are more complex and may include multiple Directus extensions united by a common logic.

I suggest to add possibility to have package.json with somesthing similar to this:
```
{
	"name": "directus-extension-myname",
	"version": "1.0.0",
	"keywords": [
		"directus",
		"directus-extension",
		"directus-custom-endpoint"
	],
	"directus:extension": [
		{
			"type": "endpoint",
			"path": "<path-to-directus-extensions>/endpoints/myendpoint/index.js",
			"source": "..",
			"host": "^9.6.0"
		},
		{
			"type": "hook",
			"path": "<path-to-directus-extensions>/hooks/myhook/index.js",
			"source": "..",
			"host": "^9.6.0"
		}
	],
	"scripts": {
		"build": "directus-extension build"
	},
	"devDependencies": {
		"@directus/extensions-sdk": "9.6.0"
	},
	"dependencies": {}
}
```

**Comment by rijkvanzanten** [comment]
Totally! Directus currently already checks against a bundle-type extension, but that's not yet documented as we're still wrapping up the details. @nickrum could you shed some light on how it currently works?

**Reply by thefairman** [reply]
has there been any information about this feature somewhere?

**Reply by tillsanders** [reply]
@rijkvanzanten @nickrum I'm also looking for information / updates regarding the bundle-type extension. Any hints?

**Reply by nickrum** [reply]
The extension type @rijkvanzanten is talking about is actually called `pack` and might not be as useful in this case as it only serves as a virtual container for other package extensions that have been published to npm.

That being said there is a new `bundle` type in the works that will allow you to bundle multiple extensions into single App and API entrypoints. The relevant PR is waiting to be reviewed and merged: #15672

**Comment by d1rOn** [comment]
I found solution for myself: to use lerna with packages which are all exporting themselves to parent directory, where Directus installed.

```
<DirectusRoot>/src — path of lerna
<DirectusRoot>/src/packages/hooks/my-hook
<DirectusRoot>/src/packages/endpoints/my-endpoint
… etc …

```
and in package.json of each separate package I have something like this:

```
"directus:extension": {
    "type": "hook",
    "path": "../../../../extensions/hooks/my-hook/index.js",
    "source": "src/index.js",
    "host": "^9.11.1"
  }
```

**Reply by thefairman** [reply]
thanks for the hint about lerna, I'll see what it is :)

**Comment by cedricDevWP** [comment]
do you have a solution that is directly integrated into the package (ex: package.json)?

**Comment by nickrum** [comment]
Support for this has been added as part of `bundle` extensions.
